xdsl/interpreters/linalg.py

```diff
@@ -162,3 +163,96 @@ def run_mat_mul(
         if len(op.results) > 0:
             return (res,)
         return ()
+
+    @impl(linalg.PoolingNchwMaxOp)
+    def run_pooling_nchw_max(
+        self,
+        interpreter: Interpreter,
+        op: linalg.PoolingNchwMaxOp,
+        args: tuple[Any, ...],
+    ) -> tuple[Any, ...]:
+        input, kernel_filter, res = args[0], args[1], args[2]
+        assert isinstance(input, ShapedArray)
+        assert isinstance(kernel_filter, ShapedArray)
+        assert isinstance(res, ShapedArray)
+        input = cast(ShapedArray[float], input)
+        kernel_filter = cast(ShapedArray[float], kernel_filter)
+        res = cast(ShapedArray[float], res)
+        assert all(res.data_ptr[i] == 0.0 for i in range(len(res.data)))
```

Comment:
```suggestion
        assert all(res.data_ptr[i] == 0.0 for i in range(len(res.data)))
```
Maybe raising a NotImplementedError here would be better to make it clear that it's not a case we support yet